pkg/distributor/distributor.go

```diff
@@ -862,6 +880,9 @@ func (d *Distributor) rateLimit(tenantID string, req *distributormodel.PushReque
 	// rate limit the request
 	if !d.ingestionRateLimiter.AllowN(time.Now(), tenantID, int(req.TotalBytesUncompressed)) {
 		validation.DiscardedProfiles.WithLabelValues(string(validation.RateLimited), tenantID).Add(float64(req.TotalProfiles))
+
+		// note(bryanhuhta): We don't need to label this metric with service
+		// name as rate limited requests don't count towards billable bytes.
 		validation.DiscardedBytes.WithLabelValues(string(validation.RateLimited), tenantID).Add(float64(req.TotalBytesUncompressed))
```

Comment:
I think this is a valid assumption. When calculating billable bytes, the recording rules seem to subtract out any bytes that were rate limited.

https://github.com/grafana/deployment_tools/blob/6b6e9832e7abb0fa2b8aeafc3aa6ef75d69880f6/ksonnet/lib/billing-mixin/recording_rules/profiles.libsonnet#L48-L50